File: src/main/java/com/qa/Salesforce/pages/HomePage.java
package com.qa.Salesforce.pages;

import com.microsoft.playwright.Page;

public class HomePage {
	private Page page;

	// String locators
	String objSelector = "//span[text()='Show Navigation Menu']";
	String Casesobj = "//span[text()='Cases']";
	String Opportunitiesobj = "//span[text()='Opportunities']";
	String Accountsobj = "//span[text()='Accounts']";
	String Leadsobj = "//span[text()='Leads']";

	// Page constructor
	public HomePage(Page page) {
		this.page = page;
	}

	// Page actions
	public void selectobj() {
		page.click(objSelector);
	}

	public void Cases() {
		page.click(Casesobj);
	}

	public Opportunity Opportunities() {
		page.click(Opportunitiesobj);
		return new Opportunity(page);
	}

	public void Leads() {
		page.click(Leadsobj);
	}

	public void Accounts() {
		page.click(Accountsobj);
	}

}

File: src/main/java/com/qa/Salesforce/pages/HomePage.js
// package com.qa.Salesforce.pages; // Removed the package declaration as this is Javascript

class HomePage {
    constructor(page) {
      this.page = page;
      this.objSelector = page.locator('//span[text()="ShowNavigationMenu"]');
      this.Casesobj = page.locator('//span[text()="Cases"]');
      this.Opportunitiesobj = page.locator('//span[text()="Opportunities"]');
      this.Accountsobj = page.locator('//span[text()="Accounts"]');
      this.Leadsobj = page.locator('//span[text()="Leads"]');
    }
  
    async selectobj() {
      await this.page.click(this.objSelector);
    }
  
    async Cases() {
      await this.page.click(this.Casesobj);
    }
  
    async Opportunities() {
      await this.page.click(this.Opportunitiesobj);
    }
  
    async Leads() {
      await this.page.click(this.Leadsobj);
    }
  
    async Accounts() {
      await this.page.click(this.Accountsobj);
    }
  }
  
  module.exports = HomePage;